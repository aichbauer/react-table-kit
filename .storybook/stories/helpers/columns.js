export const simpleColumns = [
  {
    header: '#',
    accessor: 'number',
  },
  {
    header: 'First',
    accessor: 'first',
  },
  {
    header: 'Last',
    accessor: 'last',
  },
  {
    header: 'Handle',
    accessor: 'handle',
  },
];

export const sortColumns = [
  {
    header: '#',
    accessor: 'number',
    sort: true,
  },
  {
    header: 'First',
    accessor: 'first',
    sort: true,
  },
  {
    header: 'Last',
    accessor: 'last',
    sort: true,
  },
  {
    header: 'Handle',
    accessor: 'handle',
    sort: true,
  },
];

export const searchColumnsInclude = [
  {
    header: '#',
    accessor: 'number',
    search: 'includes',
  },
  {
    header: 'First',
    accessor: 'first',
    search: 'includes',
  },
  {
    header: 'Last',
    accessor: 'last',
    search: 'includes',
  },
  {
    header: 'Handle',
    accessor: 'handle',
    search: 'includes',
  },
];

export const searchColumnsExact = [
  {
    header: '#',
    accessor: 'number',
    search: 'exact',
  },
  {
    header: 'First',
    accessor: 'first',
    search: 'exact',
  },
  {
    header: 'Last',
    accessor: 'last',
    search: 'exact',
  },
  {
    header: 'Handle',
    accessor: 'handle',
    search: 'exact',
  },
];
