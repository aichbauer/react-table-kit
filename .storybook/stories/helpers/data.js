import faker from 'faker';

export const data = [
  {
    number: 1,
    first: 'Mark',
    last: 'Otto',
    handle: '@mdo',
  },
  {
    number: 2,
    first: 'Jacob',
    last: 'Thornton',
    handle: '@fat',
  },
  {
    number: 3,
    first: 'Larry',
    last: 'the Bird',
    handle: '@twitter',
  },
];

export const dataOneHundred = () => {
  const array =  Array.apply(null, Array(100));
  const myArray = [];

  array.forEach((el, i) => {
    myArray.push({
    number: i+1,
    first: faker.name.firstName(i%2),
    last: faker.name.lastName(i%2),
    handle: `@${faker.company.bsBuzz()}`,
    });
  });

  return myArray;
}
