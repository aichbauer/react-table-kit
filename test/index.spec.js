import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import TableComponent from '../src';

import { simpleColumns } from './fixtures/columns';
import { data } from './fixtures/data';

configure({ adapter: new Adapter() });

test('render TableProvider', () => {
  const myComponent = (<TableComponent
    data={data}
    columns={simpleColumns}
  />);

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  const state = wrapper.state();

  expect(state.buttonExportCSV).toEqual(<button value="Export CSV" />);
  expect(state.columns).toEqual(simpleColumns);
  expect(state.data).toEqual(data);
  expect(state.exportCSV).toBe(false);
  expect(state.inputColumnSearch).toEqual(<input type="text" placeholder="Search column" />);
  expect(state.inputTableSearch).toEqual(<input type="text" placeholder="Search table" />);
  expect(state.table).toEqual(<table />);
  expect(state.tableSearch).toBe('');
  expect(state.td).toEqual(<td />);
  expect(state.th).toEqual(<th />);
  expect(state.trHead).toEqual(<tr />);
  expect(state.trBody).toEqual(<tr />);
  expect(state.search).toEqual({});
  expect(state.sort).toEqual({});
  expect(tree).toMatchSnapshot();
});
