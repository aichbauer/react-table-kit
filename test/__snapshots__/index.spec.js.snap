// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TableProvider 1`] = `
<div>
  
  <table>
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          First
        </th>
        <th>
          Last
        </th>
        <th>
          Handle
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="1,Mark,Otto,@mdo-body-row"
        onClick={[Function]}
      >
        <td
          onClick={[Function]}
        >
          1
        </td>
        <td
          onClick={[Function]}
        >
          Mark
        </td>
        <td
          onClick={[Function]}
        >
          Otto
        </td>
        <td
          onClick={[Function]}
        >
          @mdo
        </td>
      </tr>
      <tr
        className="2,Jacob,Thornton,@fat-body-row"
        onClick={[Function]}
      >
        <td
          onClick={[Function]}
        >
          2
        </td>
        <td
          onClick={[Function]}
        >
          Jacob
        </td>
        <td
          onClick={[Function]}
        >
          Thornton
        </td>
        <td
          onClick={[Function]}
        >
          @fat
        </td>
      </tr>
      <tr
        className="3,Larry,the Bird,@twitter-body-row"
        onClick={[Function]}
      >
        <td
          onClick={[Function]}
        >
          3
        </td>
        <td
          onClick={[Function]}
        >
          Larry
        </td>
        <td
          onClick={[Function]}
        >
          the Bird
        </td>
        <td
          onClick={[Function]}
        >
          @twitter
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
